src/main/scala/za/co/absa/spark/partition/sizing/sizer/FromDataframeSampleSizer.scala

```diff
@@ -16,22 +16,30 @@
 
 package za.co.absa.spark.partition.sizing.sizer
 
+import jdk.jfr.Experimental
 import org.apache.spark.sql.DataFrame
 import za.co.absa.spark.partition.sizing.RecordSizer
 import za.co.absa.spark.partition.sizing.types.ByteSize
 import za.co.absa.spark.partition.sizing.utils.RowSizer
 
+@Experimental
 class FromDataframeSampleSizer(sampleSize: Int = 1) extends RecordSizer {
 
   override def performRowSizing(df: DataFrame): ByteSize = {
     if(df.isEmpty) {
       0L
     } else {
-      val (rowsTotalSize, rowCount) = df.head(sampleSize) //TODO head could be skewed
-        .foldLeft(0L, 0L) { case ((size, count), row) =>
-          (size + RowSizer.rowSize(row), count + 1)
-        }
-      ceilDiv(rowsTotalSize, rowCount)
+      val count = df.count()
```

Comment:
One of the reasons I suggested the above (going with fraction instead of exact numbers) is to avoid the `df.count()`, which might actually (partially) counteract the point of taking just a sample of the data.
Also it can be misleading this way in logic, as the sampling doesn't give necessary the exact number of records, especially in case of bigger data sets.

Reply:
I agree that recoving df.count() is a good idea, but I am not sure about the percentage, since taking 10% of a huge dataset could still be quite expensive